### PR TITLE
Fix #10538: Android+WebGL PS struct variables has mediump precision

### PR DESCRIPTION
**Pull request by jk20012001**
Re: cocos-creator/3d-tasks#

Changelog:
 *

### CODE REVIEW

**Comment by jk20012001** [comment]
https://github.com/cocos-creator/engine/pull/9863

**Comment by jk20012001** [comment]
https://github.com/cocos-creator/3d-tasks/issues/10538

File: editor/assets/effects/pipeline/deferred-lighting.effect [hunk 1]
```diff
@@ -94,18 +94,19 @@ CCProgram lighting-fs %{
     #endif
 
     s.albedo = albedoMap;
-    s.position = positionMap.xyz;
+    vec3 position = positionMap.xyz;
+    s.position = position;
```
Comment (YunHsiao) [inline editor/assets/effects/pipeline/deferred-lighting.effect]: HIGHP_VALUE_TO_STRUCT_DEFINED(position, s.position);

**Review by pandamicro — APPROVED** [review]

File: cocos/core/pipeline/render-pipeline.ts [hunk 2]
```diff
@@ -47,6 +48,8 @@ import { PipelineUBO } from './pipeline-ubo';
 import { RenderFlow } from './render-flow';
 import { IPipelineEvent, PipelineEventProcessor, PipelineEventType } from './pipeline-event';
 import { decideProfilerCamera } from './pipeline-funcs';
+import { OS } from '../../../pal/system-info/enum-type';
+import { macro } from '..';
```
Comment (pandamicro) [inline cocos/core/pipeline/render-pipeline.ts]: should import from exact script position

File: editor/assets/effects/pipeline/deferred-lighting.effect (hunk 1 above)
Comment (jk20012001) [reply editor/assets/effects/pipeline/deferred-lighting.effect]: careful still you careful
image

**Review by YunHsiao — APPROVED** [review]

**Pull request merged**